### PR TITLE
Google.Analytics (major) update to version 4

### PR DESCRIPTION
**Pull request by ZbynekPelunek**
This is an upgrade from Google Analytics 3 to 4. As found out, Google completely changed a lot of things and made a public REST API which is called `v1beta` or `v1alpha` . The Beta is considered stable with no possible breaking changes.

The upgrade will remove old components and add new ones:

- [x] Create Events (this one is not using `v1beta` route)
- [x] Generate Report

### CODE REVIEW

**Comment by ZbynekPelunek** [comment]
When you create an event using the Create Event component, you can find the event on GA4 website here:
Screenshot 2024-11-18 115906

File: src/appmixer/google/analytics/CreateEvents/CreateEvents.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        }
+
+        const qeuryParams = {
+            measurement_id: measurementId, api_secret: apiSecret
+        };
+        const body = {
+            client_id: clientId, events: transformedEvents
+        };
+
+        await context.httpRequest({
```
Comment (vtalas) [inline src/appmixer/google/analytics/CreateEvents/CreateEvents.js:37]: there has to be the `context.sendJson`, even with empty object. without it, the flow will stuck on the CreateEvent component

File: src/appmixer/google/analytics/CreateEvents/component.json [hunk 2]
```diff
@@ -1,134 +1,118 @@
...
-        ]
-    },
-    "quota": {
-        "manager": "appmixer:google:analytics",
-        "maxWait": 5000,
-        "resources": "requests",
-        "scope": {
-            "userId": "{{userId}}"
-        }
+        "service": "appmixer:google:analytics"
```
Comment (vtalas) [inline src/appmixer/google/analytics/CreateEvents/component.json:6]: is the authentication needed? there is no authenticated call in the implementation

File: src/appmixer/google/analytics/CreateEvents/CreateEvents.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+            throw new context.CancelError('Request can have a maximum of 25 events');
+        }
+
+        const errors = validateEvents(transformedEvents);
+
+        if (errors.length > 0) {
+            throw new context.CancelError(errors);
+        }
+
+        const qeuryParams = {
```
Comment (vtalas) [inline src/appmixer/google/analytics/CreateEvents/CreateEvents.js:30]: typo

**Review by vtalas — CHANGES_REQUESTED** [review]

**Review by vtalas — APPROVED** [review]

**Pull request merged**